### ISSUE TITLE
Is there a way to dump the database to a human-readable text file?

### ISSUE THREAD

**Issue by Copyrites** [cboxdoerfer/fsearch#310]
I lost a couple of disks and it seems like my most up to date list of their contents is in fsearch's database.

I'd like to dump the database to something human readable, or at least parse-able, like csv.

Is there an easy way to do that?   Maybe you have a tool you put together for debugging that you could share?

**Comment by cboxdoerfer** [comment]
No, sorry I don't have such a tool at hand. But writing one shouldn't take long, I'll see if I can do that this weekend.

Do you only need the path+filename or also additional information like size or mtime?

**Comment by Copyrites** [comment]
That is super cool of you.  I'd like to get all the fields.  Please don't go out of your way to make it pretty (unless you want to) I'm planning on feeding it to a perl script for processing so raw mtime in seconds would actually be easier for me personally.

I don't know if it matters, but I saw your other comment about having to change the database format before releasing the 0.1 stable but my database is from the 0.1beta4 build.

**Comment by cboxdoerfer** [comment]
Ok, will do.

> I don't know if it matters, but I saw your other comment about having to change the database format before releasing the 0.1 stable but my database is from the 0.1beta4 build.

That is good to know, I'll need to base the dump code on the old format then. I'll let you know when it's done and how to build and use it.

**Comment by cboxdoerfer** [comment]
@Copyrites, so I just quickly hacked together a small dump utility. The code can be found here: https://pastebin.com/vrUXF8SN

If you're not familiar with C, you need to do the following. Download or copy the code from pastebin to a file called `fs_dump.c`, then open the terminal, navigate to the folder with the `fs_dump.c` file. You can then compile it with `gcc fs_dump.c -o fs_dump`, which will generate an executable `fs_dump`.

The `fs_dump` tool works like that `fs_dump DATABASE_FILE OUTPUT_FILE`.

`DATABASE_FILE` is an fsearch database. There might be multiple ones and they can be found here `~/.config/fsearch/database/.../database.db`. Each one holds the content of a particular location you included, e.g. `/home/user`, `/mnt/data`, ...

`OUTPUT_FILE` is simply the name of the file you want to dump the database content to.

So to summarize, it might look like that:

```
gcc fs_dump.c -o fs_dump
./fs_dump ~/.config/fsearch/SOME_SHA_HASH/database.db fs_dump.txt
```

The output file has the following format: `size, mtime, path`

If something doesn't work or you need more help, let me know!

**Comment by Copyrites** [comment]
I just gave it a try and it seems to work great.  Thanks much.  

Are you going to include it in future releases of fsearch? (or maybe add the functionality directly to fsearch?)   I imagine I'm not the only one who find it useful.

**Comment by cboxdoerfer** [comment]
Yes, I'll add that as an official feature, but probably only for the new format.

**Comment by He1my** [comment]
has it been added to the new version?
if not then is there a way to dump the database contents into a text file for the new format?

**Comment by cboxdoerfer** [comment]
> has it been added to the new version? if not then is there a way to dump the database contents into a text file for the new format?

No, it hasn't. Dumping the database to a file or the console is planned for version 0.3.
